ipa-core/src/protocol/ipa_prf/aggregation/mod.rs

```diff
@@ -395,17 +395,20 @@ pub mod tests {
                 Ok(input_row(1, &[0, 1, 0, 1, 0, 0, 0, 0])),
                 Ok(input_row(1, &[0, 0, 1, 1, 0, 0, 0, 0])),
             ];
-
             let result = TestWorld::default()
                 .upgraded_semi_honest(inputs.into_iter(), |ctx, inputs| {
                     let num_rows = inputs.len();
                     aggregate_values::<BA8, 8>(ctx, stream::iter(inputs).boxed(), num_rows)
                 })
                 .await
                 .map(Result::unwrap)
-                .reconstruct();
+                .reconstruct_arr();
 
-            assert_eq!(result, [0u32, 1, 2, 3, 0, 0, 0, 0].map(BA8::truncate_from));
+            assert_eq!(
+                result,
+                input_row(8, &[0u32, 1, 2, 3, 0, 0, 0, 0])
```

Comment:
I'm surprised you're still expecting the same thing given this method now returns a new result.

Reply:
similar comment thread above, but here `result` is vectorized still and `input_row` is vectorizing the expected result.  Previously `input_row` wouldn't have appeared here.